src/Core/Table/TableView.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Table;
+use Symfony\Component\Form\FormView;
+
+/**
+ * Class TableView is responsible for storing table data that is passed to template to render table
+ */
+final class TableView
+{
+    /**
+     * @var string
+     */
+    private $identifier;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var FormView
+     */
+    private $formView;
+
+    /**
+     * @var array
+     */
+    private $columns;
+
+    /**
+     * @var array
+     */
+    private $rows;
+
+    /**
+     * @var int
+     */
+    private $rowsTotal;
+
+    /**
+     * @param string    $identifier     Table identifier should be unique per table and will act as ID on html table element
+     * @param string    $name           Table name
+     * @param array     $columnViews    Table columns
+     * @param array     $rowViews       Table rows data
+     * @param int       $rowsTotal      Total count of all rows
+     * @param FormView  $formView       Filters form view
+     */
+    public function __construct($identifier, $name, array $columnViews, array $rowViews, $rowsTotal, FormView $formView)
```

Comment:
note that sometimes we dont need to filter data using a form in soem pages, so I guess FormView should be  optional: am I right?

Reply:
yup, ti can be optional. in any case, if will be empy form if none of the columns have FormType set.